### ISSUE TITLE
Segmentation Loss

### ISSUE THREAD

**Issue by junxnone** [junxnone/tio#590]
## Reference

- [从loss处理图像分割中类别极度不均衡的状况---keras](https://blog.csdn.net/m0_37477175/article/details/83004746)
- [An overview of semantic image segmentation.  - Defining a loss function](https://www.jeremyjordan.me/semantic-segmentation/#loss)
- [Losses for Image Segmentation](https://lars76.github.io/neural-networks/object-detection/losses-for-segmentation/)
- [医学图像分割常用的损失函数](https://blog.csdn.net/Biyoner/article/details/84728417)
- [segmentation loss summary - JunMa11](https://github.com/JunMa11/SegLoss)

## Brief
- 2016年的VNet论文首次提出了Dice Loss，应该是Class-Level的Loss的代表
- [Dice Loss](/Dice_Loss)
- [Focal Loss](/Focal_Loss)

## Log loss

多分类
![image](https://user-images.githubusercontent.com/2216970/68283948-7d559f80-00b7-11ea-9c3c-22b6cc415a9f.png)

---
二分类
![image](https://user-images.githubusercontent.com/2216970/70404440-0c6a2480-1a75-11ea-8023-d9ab382ed4b4.png)

- 每一次梯度的回传对每一个类别具有相同的关注度, 极易受到类别不平衡的影响

## WCE Loss - Weighted cross-entropy 
![image](https://user-images.githubusercontent.com/2216970/70415781-69c39d00-1a98-11ea-8c57-fc124472e7ca.png)

## Dice Loss
![image](https://user-images.githubusercontent.com/2216970/70405474-4e489a00-1a78-11ea-9e0f-3c2fd4ee5a2b.png)

- `Dice Loss = 1 - F1-Score`

## IOU Loss
![image](https://user-images.githubusercontent.com/2216970/70418534-8b278780-1a9e-11ea-91d4-155431c47caa.png)
- `I(X) = X * Y`
- `U(X) = X + Y - X * Y`

![image](https://user-images.githubusercontent.com/2216970/70418520-85ca3d00-1a9e-11ea-9bb6-fc077d269064.png)

**Comment by junxnone** [comment]
#474
